src/ansys/additive/core/additive.py

```diff
@@ -624,3 +624,26 @@ def _check_for_duplicate_id(self, inputs):
             if input.id in ids:
                 raise ValueError(f'Duplicate simulation ID "{input.id}" in input list')
             ids.append(input.id)
+
+    def _check_for_thermal_history(self, response, input) -> SingleBeadSummary:
```

Comment:
The bulk of this should be moved to `__init__()` in `SingleBeadSummary` as mentioned above.